scripts/extract-build-number.sh

```
@@ -0,0 +1,68 @@
+fi
+
+DMG_PATH="$1"
+
+if [ ! -f "$DMG_PATH" ]; then
+    echo "Error: DMG file not found: $DMG_PATH" >&2
+    exit 1
+fi
+
+# Create temporary directory for mounting
+TEMP_DIR=$(mktemp -d)
+trap "rm -rf $TEMP_DIR" EXIT
```

Shellcheck (0.10.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)